lib/wickedbase.pm

```diff
@@ -30,29 +31,37 @@ sub assert_wicked_state {
 
 sub get_ip {
     my ($self, %args) = @_;
+    my $ip;
+
     if ($args{type} eq 'host') {
-        if ($args{no_mask}) {
-            return $args{is_wicked_ref} ? '10.0.2.10' : '10.0.2.11';
-        }
-        else {
-            return $args{is_wicked_ref} ? '10.0.2.10/15' : '10.0.2.11/15';
-        }
+        $ip = $args{is_wicked_ref} ? '10.0.2.10/15' : '10.0.2.11/15';
     }
     elsif ($args{type} eq 'gre1') {
-        return $args{is_wicked_ref} ? '192.168.1.1' : '192.168.1.2';
+        $ip = $args{is_wicked_ref} ? '192.168.1.1' : '192.168.1.2';
     }
     elsif ($args{type} eq 'sit1') {
-        return $args{is_wicked_ref} ? '2001:0db8:1234::000e' : '2001:0db8:1234::000f';
+        $ip = $args{is_wicked_ref} ? '2001:0db8:1234::000e' : '2001:0db8:1234::000f';
     }
     elsif ($args{type} eq 'tunl1') {
-        return $args{is_wicked_ref} ? '3.3.3.10' : '3.3.3.11';
+        $ip = $args{is_wicked_ref} ? '3.3.3.10' : '3.3.3.11';
     }
     elsif ($args{type} eq 'tun1' || $args{type} eq 'tap1') {
-        return $args{is_wicked_ref} ? '192.168.2.10' : '192.168.2.11';
+        $ip = $args{is_wicked_ref} ? '192.168.2.10' : '192.168.2.11';
     }
     elsif ($args{type} eq 'br0') {
-        return $args{is_wicked_ref} ? '10.0.2.10' : '10.0.2.11';
+        $ip = $args{is_wicked_ref} ? '10.0.2.10' : '10.0.2.11';
+    }
+    elsif ($args{type} eq 'vlan') {
+        $ip = $args{is_wicked_ref} ? '42.42.42.10/24' : '42.42.42.11/24';
     }
+    elsif ($args{type} eq 'vlan_changed') {
+        $ip = $args{is_wicked_ref} ? '42.42.42.110/24' : '42.42.42.111/24';
+    }
+
+    if (defined($ip) && $args{no_mask}) {
+        $ip =~ s'/\d+$'';
+    }
+    return $ip if (defined($ip));
```

Comment:
hm so let's see , so if caller have something like this `my $returned_ip=get_ip();` , `$returned_ip` will be undefined , right ? and if you remove this if it will still be undefined :)

Reply:
I hat in mind "don't return undef explicit". So if someone calls it in array context, he will get a empty array, instead of a array with one value of undef.